### PR TITLE
ENH: Catch RuntimeErrors from qiita_env drop

### LINKED ISSUE #135: qiita_env drop_demo_env fails on nonexistent DB
If the demo database doesn't exist the command will fail with the following output:

```python

Traceback (most recent call last):
  File "/Users/yoshikivazquezbaeza/.virtualenvs/qiita/bin/qiita_env", line 10, in <module>
    execfile(__file__)
  File "/Users/yoshikivazquezbaeza/git_sw/qiita/scripts/qiita_env", line 113, in <module>
    qiita_env()
  File "/Users/yoshikivazquezbaeza/.virtualenvs/qiita/lib/python2.7/site-packages/click/core.py", line 384, in __call__
    return self.main(*args, **kwargs)
  File "/Users/yoshikivazquezbaeza/.virtualenvs/qiita/lib/python2.7/site-packages/click/core.py", line 370, in main
    self.invoke(ctx)
  File "/Users/yoshikivazquezbaeza/.virtualenvs/qiita/lib/python2.7/site-packages/click/core.py", line 614, in invoke
    return self.invoke_subcommand(ctx, cmd, cmd_name, ctx.args[1:])
  File "/Users/yoshikivazquezbaeza/.virtualenvs/qiita/lib/python2.7/site-packages/click/core.py", line 623, in invoke_subcommand
    return cmd.invoke(cmd_ctx)
  File "/Users/yoshikivazquezbaeza/.virtualenvs/qiita/lib/python2.7/site-packages/click/core.py", line 524, in invoke
    ctx.invoke(self.callback, **ctx.params)
  File "/Users/yoshikivazquezbaeza/.virtualenvs/qiita/lib/python2.7/site-packages/click/core.py", line 236, in invoke
    return callback(*args, **kwargs)
  File "/Users/yoshikivazquezbaeza/git_sw/qiita/scripts/qiita_env", line 109, in drop_demo_env
    drop_demo_environment(user, None, host)
  File "/Users/yoshikivazquezbaeza/git_sw/qiita/qiita_db/environment_manager.py", line 188, in drop_demo_environment
    cur.execute('DROP DATABASE qiita_demo')
psycopg2.ProgrammingError: database "qiita_demo" does not exist

```

### PR DESCRIPTION
**Pull request by ElDeveloper**
Fixes #135

### CODE REVIEW

**Comment by antgonza** [comment]
:+1:

File: scripts/qiita_env [hunk 1]
```diff
@@ -81,7 +81,11 @@ def drop(ask_for_confirmation):
     # Or figure out a way to use
     # http://click.pocoo.org/3/options/#password-prompts
     # and have it work with Travis...
-    drop_environment(ask_for_confirmation)
+    try:
+        drop_environment(ask_for_confirmation)
+    except RuntimeError as e:
+        click.echo(str(e), stderr, False)
+        exit(1)
```
Comment (wasade) [inline scripts/qiita_env]: I think this should be `env.exit(1)`

File: scripts/qiita_env [hunk 2]
```diff
@@ -81,7 +81,11 @@ def drop(ask_for_confirmation):
     # Or figure out a way to use
     # http://click.pocoo.org/3/options/#password-prompts
     # and have it work with Travis...
-    drop_environment(ask_for_confirmation)
+    try:
+        drop_environment(ask_for_confirmation)
+    except RuntimeError as e:
+        click.echo(str(e), stderr, False)
```
Comment (wasade) [inline scripts/qiita_env]: in click 3.0, you can do `click.echo(str(e), err=True)`

**Comment by wasade** [comment]
just a couple of commens

**Comment by ElDeveloper** [comment]
Fixed. @wasade, the solution was actually cleaner ... click doesn't cease to surprise me.

**Comment by antgonza** [comment]
flake8 errors.

**Comment by wdwvt1** [comment]
:+1: 
@ElDeveloper - looks good, click error is now raised when attempting to drop an unknown database. I think you could make the error message a bit cleaner if we were being nitpicky. Currently, we get an error like:
`Error: Cannot connect to database: FATAL:  database "already_dropped_database" does not exist`
We could parse this and give something like below which is just a bit cleaner but perhaps is less informative for someone who knows what the syntax indicates. 
`Error: cannot connect to database 'already_dropped_database'.`

**Comment by ElDeveloper** [comment]
Thanks @wdwvt1.

**Pull request merged**